tools/generate-readme.js

```diff
@@ -37,23 +37,30 @@ const FOLDER = path.resolve('src/js/components');
 
 components(FOLDER).forEach(component => {
   /* eslint-disable */
-  const { doc, themeDoc } = require(path.join(FOLDER, component, 'doc.js'));
-  const componentModule = require(path.join(FOLDER, component, 'index.js'));
-  // we use the second array element since the first is '__esModule'.
-  const Component =
-    componentModule[
-      Object.keys(componentModule).filter(k => k === component)[0]
-    ];
-  /* eslint-enable */
-
-  const readmeDestination = path.join(FOLDER, component, 'README.md');
-
-  const DocumentedComponent = doc(Component);
-
-  const readmeContent = themeDoc
-    ? `${replaceHoc(DocumentedComponent.toMarkdown())}\n${toMarkdown(themeDoc)}`
-    : `${replaceHoc(DocumentedComponent.toMarkdown())}`;
-  del(readmeDestination).then(() =>
-    fs.writeFileSync(readmeDestination, readmeContent),
-  );
+  try {
+    const { doc, themeDoc } = require(path.join(FOLDER, component, 'doc.js'));
+    const componentModule = require(path.join(FOLDER, component, 'index.js'));
+
+    // we use the second array element since the first is '__esModule'.
+    const Component =
+      componentModule[
+        Object.keys(componentModule).filter(k => k === component)[0]
+      ];
+    /* eslint-enable */
+
+    const readmeDestination = path.join(FOLDER, component, 'README.md');
+
+    const DocumentedComponent = doc(Component);
+
+    const readmeContent = themeDoc
+      ? `${replaceHoc(DocumentedComponent.toMarkdown())}\n${toMarkdown(
+          themeDoc,
+        )}`
+      : `${replaceHoc(DocumentedComponent.toMarkdown())}`;
+    del(readmeDestination).then(() =>
+      fs.writeFileSync(readmeDestination, readmeContent),
+    );
+  } catch (ex) {
+    // don't make changes to README.md if doc.js file isn't found
```

Comment:
Are we intentionally hiding the exception? shouldn't we print it to the console in case something will go wrong?

Reply:
As components have react-desc removed and no longer have doc files, the catch block will trigger. In this case nothing went wrong, it was expected that as react-desc is removed components will no longer have doc files in grommet. So I don't think printing to console is the right thing to do since it will make it look like something went wrong. This block was just added to maintain backwards compatibility for components that still use react-desc. Once react-desc is removed it can be removed as well so it is just temporary.

Reply:
Please add this detailed response as a comment on the catch block to avoid future confusion.